src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReader.cs

```diff
@@ -1712,32 +1734,37 @@ internal virtual IDtdInfo DtdInfo
 
         internal static ConformanceLevel GetV1ConformanceLevel(XmlReader reader)
         {
-            XmlTextReaderImpl tri = GetXmlTextReaderImpl(reader);
+            XmlTextReaderImpl? tri = GetXmlTextReaderImpl(reader);
             return tri != null ? tri.V1ComformanceLevel : ConformanceLevel.Document;
         }
 
-        private static XmlTextReaderImpl GetXmlTextReaderImpl(XmlReader reader)
+        private static XmlTextReaderImpl? GetXmlTextReaderImpl(XmlReader reader)
         {
-            XmlTextReaderImpl tri = reader as XmlTextReaderImpl;
+            XmlTextReaderImpl? tri = reader as XmlTextReaderImpl;
```

Comment:
Nit Same here and below
```suggestion
           if (reader is XmlTextReaderImpl tri)
```